### PR TITLE
Remove placeholder from `WP_Term_Query` cache key

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/57298

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-term-query.php [hunk 1]
```diff
@@ -1142,4 +1131,37 @@ protected function populate_terms( $terms ) {
...
 	}
+
+	/**
+	 * Generate cache key.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param array  $args Query arguments.
```
Comment (peterwilsoncc) [inline src/wp-includes/class-wp-term-query.php]:
```suggestion
	 * @param array  $args WP_Term_Query arguments.
```

**Review by peterwilsoncc — COMMENTED** [review]
LGTM. 

I got a review ping while the PR is in draft, I presume that's for unit tests to come. They'd be lovely :) 

Thanks

**Comment by spacedmonkey** [comment]
Unit tests added.

File: tests/phpunit/tests/term/query.php [hunk 2]
```diff
@@ -867,4 +867,114 @@ public function test_hide_empty_should_include_empty_parents_of_nonempty_childre
...
+
+		$query_vars = $query1->query_vars;
+		$request    = $query1->request;
+
+		$reflection = new ReflectionMethod( $query1, 'generate_cache_key' );
+		$reflection->setAccessible( true );
+
+		$cache_key_1 = $reflection->invoke( $query1, $query_vars, $request );
+
+		$request_with_placeholder = $wpdb->remove_placeholder_escape( $request );
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/term/query.php]: Should this be `_without_`? My reading of `remove_placeholder_escape` is that it replaces the placeholder with the `%` symbol

**Review by peterwilsoncc — COMMENTED** [review]
LGTM, one question about a variable name is all.

File: tests/phpunit/tests/term/query.php (hunk 2 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/term/query.php]: Fixed

**Comment by spacedmonkey** [comment]
Committed. https://github.com/WordPress/wordpress-develop/commit/ec6b2f20455cf39bd269b2f71749c03140d79ef1

**Pull request closed without merging**